app/services/ClaimToAdjustService.scala

```diff
@@ -86,4 +89,67 @@ class ClaimToAdjustService @Inject()(val financialDetailsConnector: FinancialDet
     }).map(_.flatten)
   }
 
+  def getPoaTaxYearForEntryPoint(nino: Nino)(implicit hc: HeaderCarrier, user: MtdItUser[_]): Future[Either[Throwable, Option[TaxYear]]] = {
+    checkCrystallisation(nino, getPoaAdjustableTaxYears).flatMap {
```

Comment:
I would suspect that for-comprehension is a good option to use here instead of pattern matching result evertime